programs/clearing_house/src/math/amm.rs

```diff
@@ -189,35 +191,13 @@ pub fn calculate_new_oracle_price_twap(
             .ok_or_else(math_error!())?,
     ))?;
     let from_start = max(
-        1,
+        0,
         cast_to_i128(amm.funding_period)?
             .checked_sub(since_last)
             .ok_or_else(math_error!())?,
     );
 
-    // ensure amm.last_oracle_price is proper
-    let capped_last_oracle_price = if amm.last_oracle_price > 0 {
-        amm.last_oracle_price
-    } else {
-        oracle_price
-    };
-
-    // nudge last_oracle_price up to .1% toward oracle price
-    let capped_last_oracle_price_10bp = capped_last_oracle_price
-        .checked_div(1000)
-        .ok_or_else(math_error!())?;
-
-    let mut interpolated_oracle_price = min(
-        capped_last_oracle_price
-            .checked_add(capped_last_oracle_price_10bp)
-            .ok_or_else(math_error!())?,
-        max(
-            capped_last_oracle_price
-                .checked_sub(capped_last_oracle_price_10bp)
-                .ok_or_else(math_error!())?,
-            oracle_price,
-        ),
-    );
+    let mut interpolated_oracle_price = oracle_price;
```

Comment:
nitpick: you can init this in one line. if you like the other way better feel free to keep it

```
    // if an oracle delay impacted last oracle_twap, shrink toward mark_twap
    let interpolated_oracle_price = if amm.last_mark_price_twap_ts > amm.last_oracle_price_twap_ts {
        let since_last_valid = cast_to_i128(
            amm.last_mark_price_twap_ts
                .checked_sub(amm.last_oracle_price_twap_ts)
                .ok_or_else(math_error!())?,
        )?;
        msg!(
            "correcting oracle twap update (oracle previously invalid for {:?} seconds)",
            since_last_valid
        );

        let from_start_valid = max(
            1,
            cast_to_i128(amm.funding_period)?
                .checked_sub(since_last_valid)
                .ok_or_else(math_error!())?,
        );
        calculate_weighted_average(
            cast_to_i128(amm.last_mark_price_twap)?,
            oracle_price,
            since_last_valid,
            from_start_valid,
        )?
    } else {
        oracle_price
    };